bootstrap-app/src/cmr/bootstrap/services/dispatch/core_async_dispatch.clj

```diff
@@ -0,0 +1,143 @@
+(ns cmr.bootstrap.services.dispatch.core-async-dispatch
```

Comment:
Ah, so it seems that you and I have a similar philosophy about this :-) I gave pretty middle-ground Clojure example -- _but_ :-) how I usually do this, when no one's looking, is define my protocols in `proj.some.ns.protocols` then require it as well as use potemkin in `proj.some.ns.thing.core` and then put implementations in `proj.some.ns.thing.impl.*`. Which I think is kind of what you were getting at ...? (basically creating protocols like an interface in Java).